staging/src/k8s.io/apimachinery/pkg/fields/selector.go

```diff
@@ -77,7 +78,7 @@ func (t *hasTerm) RequiresExactMatch(field string) (value string, found bool) {
 func (t *hasTerm) Transform(fn TransformFunc) (Selector, error) {
 	field, value, err := fn(t.field, t.value)
 	if err != nil {
-		return nil, err
+		return nil, errors.ErrorToBadRequest(err)
```

Comment:
This is not what I would expected.  Generic field selectors have nothing to do with API errors.  What are you trying to accomplish here?